mods/default/nodes.lua

```diff
@@ -586,30 +586,116 @@ minetest.register_node("default:snow", {
 	inventory_image = "default_snowball.png",
 	wield_image = "default_snowball.png",
 	paramtype = "light",
+	paramtype2 = "leveled",
 	buildable_to = true,
 	floodable = true,
+	walkable = true,
 	drawtype = "nodebox",
+	leveled = 16,
 	node_box = {
-		type = "fixed",
-		fixed = {
-			{-0.5, -0.5, -0.5, 0.5, -0.25, 0.5},
-		},
+		type = "leveled",
+		fixed = {-0.5, -0.5, -0.5, 0.5, -0.5, 0.5},
 	},
 	collision_box = {
-		type = "fixed",
-		fixed = {
-			{-0.5, -0.5, -0.5, 0.5, -7 / 16, 0.5},
-		},
+		type = "leveled",
+		fixed = {-0.5, -0.5, -0.5, 0.5, -0.5, 0.5},
+	},
+	selection_box = {
+		type = "leveled",
+		fixed = {-0.5, -0.5, -0.5, 0.5, -0.5, 0.5},
 	},
 	groups = {crumbly = 3, falling_node = 1, snowy = 1},
 	sounds = default.node_sound_snow_defaults(),
+	node_placement_prediction = nil,
 
 	on_construct = function(pos)
-		pos.y = pos.y - 1
-		if minetest.get_node(pos).name == "default:dirt_with_grass" then
-			minetest.set_node(pos, {name = "default:dirt_with_snow"})
+		pos.y = pos.y-1
+		local node = minetest.get_node(pos)
+		if node.name == "default:dirt"
+		or node.name == "default:dirt_with_grass"
+		or node.name == "default:dirt_with_dry_grass"
+		or node.name == "default:dirt_with_grass_footsteps"
+		or node.name == "default:dirt_with_rainforest_litter"
+		or node.name == "default:dirt_with_coniferous_litter" then
+			node.name = "default:dirt_with_snow"
+			minetest.set_node(pos, node)
+		end
+	end,
+
+	-- Handle node drops due to node level.
+	on_dig = function(pos, node, digger)
+		local level = minetest.get_node_level(pos)
+		minetest.node_dig(pos, node, digger)
+		if minetest.get_node(pos).name ~= node.name then
+			local inv = digger:get_inventory()
+			if not inv then
+				return
+			end
+			local left = inv:add_item("main", "default:snow "..tostring(level/16))
+			if not left:is_empty() then
+				minetest.add_item({
+					x = pos.x + math.random()/2-0.25,
+					y = pos.y + math.random()/2-0.25,
+					z = pos.z + math.random()/2-0.25,
+				}, left)
+			end
```

Comment:
Use [`minetest.handle_node_drops`](https://github.com/minetest/minetest/blob/9b067efb99c7b8682639736e7c4908b2e3e6e524/doc/lua_api.txt#L4503-L4508). (Eg. item drop mods need this. But this is also for consistency.)

Reply:
Where do I put it? We removed that `if` condition in one of the reviews below.

Reply:
Do something like this:
```lua
	if creative and creative.is_enabled_for and creative.is_enabled_for(digger and digger:get_player_name() or "") or
			minetest.get_node(pos).name == node.name then
		return
	end

	local remaining = ItemStack("default:snow " .. tostring(level / 16 - 1))
	if not remaining:is_empty() then
		minetest.handle_node_drops(pos, {remaining}, digger)
	end
end,
```
(And get the level before calling `node_dig`, of course.)

Reply:
Done!